.bit/responses/2.3-Week 2 Step 3.md

```diff
@@ -1,48 +1,226 @@
+
+> [Is JavaScript Synchronous or Asynchronous? What the Heck is a Promise?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+> [Master the JavaScript Interview: What is a Promise?](https://medium.com/better-programming/is-javascript-synchronous-or-asynchronous-what-the-hell-is-a-promise-7aa9dd8f3bfb)
+
+</br>
 </details>
-<br>
+
+<details>
+<summary> :exclamation: How should I define the keys to authenticate my request?</summary>
+</br>
+This function takes in one parameter, `img`, that contains the image we're trying to analyze.  Inside, we have two variables involved in the call: `subscriptionKey`  and `uriBase`.  
+
+The `process.env` object allows you to access super-secret values in your backend. This prevents hackers from getting your keys and doing bad stuff (like exploiting your resources).
+
+```js
+async function analyzeImage(img){
+    const subscriptionKey = process.env.SUBSCRIPTIONKEY;
+    const uriBase = process.env.ENDPOINT + '/face/v1.0/detect';
+}
+```
+
+**[Where](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings?tabs=portal) do you add the values?**
+
+Head to your Function App and click on `Configuration.`
+![image](https://user-images.githubusercontent.com/69332964/117517193-345eff80-af69-11eb-9284-5e5c2380f382.png)
+
+
+Then, let's add the secrets by clicking `New application setting`.
+![image](https://user-images.githubusercontent.com/69332964/117517158-18f3f480-af69-11eb-8efa-579063ad7d00.png)
+
+Let's hope you have your subscription keys and endpoint handy. Add them in as `SUBSCRIPTIONKEY` and `ENDPOINT`.
+
+> Congrats! Your keys are now safe.
+</details>
+
+
 <details>
-<summary>❓ Where do I begin?</summary>
+<summary>❗️ Specify the parameters for your request!</summary>
 </br>
 
-1. Log into your Azure portal, press **Create a Resource**, the **AI + Machine Learning** tab on the left, and finally select **Face** and fill out the necessary information.Record and save the API endpoint and [subscription key](https://docs.microsoft.com/en-us/azure/api-management/api-management-subscriptions)
+In order to specify all of our parameters easily, we're going to create a new `URLSearchParams`  object. Here's the object declared for you. I've also already specified one parameter, `returnFaceId`,  as `true` to provide an example. Add in a new parameter that requests emotion.
+
+```js
+let params = new URLSearchParams({
+	'returnFaceId': 'true',
+	'<PARAMETER NAME>': '<PARAMETER VALUE>'     //FILL IN THIS LINE
+})
+```
+
 </details>
 <br>
 
+#### 🐕 Using `Node-Fetch`
+
+In order to make POST request, we'll be using the `node-fetch` package. [What is the node-fetch package?](https://www.npmjs.com/package/node-fetch) Install it!
```

Comment:
We need to specify WHY we're making a POST request, with a reasoning behind it so it doesn't seem random